get_http.go

```diff
@@ -117,13 +118,54 @@ func (g *HttpGetter) GetFile(dst string, u *url.URL) error {
 		g.Client = httpClient
 	}
 
-	resp, err := g.Client.Get(u.String())
+	// check to see whether user has specified a range of bytes to download.
+	// if user has, but the range is invalid, fall back to downloading the
+	// whole file
+	byteRange, rangeErr := GetByteRange(u)
+	if rangeErr != nil {
+		if strings.Compare("No byte range provided", rangeErr.Error()) != 0 {
+			fmt.Printf("%s; going to download entire file without a range request",
+				rangeErr)
+		}
+	}
+
+	req, err := http.NewRequest("HEAD", u.String(), nil)
+
+	isPartialDownload := false
+	if rangeErr == nil {
+		// We first make a HEAD request so we can check if the server supports
+		// range queries. If the server/URL doesn't support HEAD requests,
+		// we just fall back to GET.
+
+		if err != nil {
```

Comment:
I'd move this entire `err` check up to when `NewRequest` is called. Under the current logic I think there is a edge case where: 

  * No byte range: `rangeErr != nil` under current logic
  * Request allocation fails `err != nil` 
  
And no error is reported since we will do a `g.Client.Do` on the GET below.

Reply:
oh yeah you're right.  That must have been an accident of refactoring because it makes no sense where I have it now.